enhancements/builds/disable-jenkins-pipeline.md

```diff
@@ -0,0 +1,249 @@
+---
+title: disable-jenkins-pipeline
+authors:
+  - "@adambkaplan"
+reviewers:
+  - @gabemontero
+  - Akram
+approvers:
+  - "@bparees"
+  - Jessica Forrester
+  - "@sbose78"
+creation-date: 2021-04-21
+last-updated: 2021-04-23
+status: provisional
+see-also: []
+replaces: []
+superseded-by: []
+---
+
+# Disable Jenkins Pipeline Strategy
+
+**Note - this enhancement is provisional and is intended to solicit feedback**
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+OpenShift 3.0 included a way to invoke a Jenkins pipeline from an OpenShift build.
+The status of the build on the OpenShift cluster would reflect the state of the pipeline in Jenkins.
+This was the first feature to support continuous integration/delivery processes on OpenShift.
+
+As OpenShift and Kubernetes have evolved, so has cloud-native CI/CD.
+Tekton and its downstream distribution - OpenShift Pipelines - provide a way of running CI/CD processes natively on Kubernetes.
+The Jenkins pipeline strategy was officially deprecated in OpenShift 4.1, but at that time there was no meaningful replacement.
+We now have that meaningful replacement with the general availability of OpenShift Pipelines.
+It is time to disable OpenShift's direct integration with Jenkins.
+
+## Motivation
+
+Distrbuting and supporting Jenkins directly has always been challenging for OpenShift.
+Jenkins is based in Java and is extended by a convoluted ecosystem of plugins, which are a constant source of bugs and CVEs.
+Historically, the sprawling nature of Jenkins and its dependencies has made it impossible for Red Hat to independently compile Jenkins and the libraries used by OpenShift's plugin extensions.
+We instead distribute Jenkins directly from community binaries, making it vulnerable to supply chain attacks from sophisticated attackers.
+Given the increasing prevalence of supply chain attacks globally (SolarWinds), relying on third parties for binary code is becoming less tolerable for product security. 
+
+Jenkins itself is not native to Kubernetes and does not use a Java runtime that has been optimized for cloud-native environments (i.e. Quarkus).
+As a result, it has poor resilience when run on Kubernetes.
+The Jenkins templates do not run Jenkins in a highly available mode, leaving its primary instance vulnerable to downtime from slow restarts.
+The Jenkins container image is also very large - it is in fact the largest image in the OpenShift payload today.
+Removing Jenkins from the OpenShift payload has been a longstanding goal that can reduce the time needed to pull the OCP payload at install/updgrade time.
+
```

Comment:
Why does removing Jenkins from payload requires removing the pipeline strategy from builds?

Reply:
strictly speaking, it does not, however, if we remove it from the payload but keep the strategy:

1) customers will need to setup their own jenkins imagestream pointing to an appropriate image before their pipelines will work
2) as long as we keep the pipeline strategy alive, we need to keep the jenkins sync plugin up to date in an available jenkins image (sync plugin is necessary for the pipeline strategy to work)

(for the record i agree with your concerns about disabling the strategy, we need to do some due diligence here to figure out who is still using the strategy and what the impact would be on them if we disabled it)

Reply:
part of moving it out of the payload to a new cpaas based image hosted on registry.redhat.io means you could still ship a samples jenkins imagestream and template, but remove the samples operator function that replaces the current image ref with the payload image.

in that sense then, jenkins would be just like all our other samples from SCL and middleware 

no need for users to set up the imagestream / template on their own

those templates and imagestreams still go through the process of being hosted in their "upstream" repo (in the jenkins case, https://github.com/openshift/jenkins) and then get pulled into https://github.com/openshift/library

Reply:
> part of moving it out of the payload to a new cpaas based image hosted on registry.redhat.io means you could still ship a samples jenkins imagestream and template, but remove the samples operator function that replaces the current image ref with the payload image.

that's fair, though it doesn't work as cleanly for disconnected installs. and it loses the tight coupling between the jenkins image version and the ocp version (possibly not that critical any more if things have stabilized)

Reply:
I'm much more comfortable with this approach despite the drawbacks that @bparees mentioned. I see it as an acceptable loss compared to removing the strategy altogether at this point.

Reply:
Removing Jenkins from the payload is being considered in a separate proposal - #841